src/clr-core/modal/modal.element.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {
+  applyMixins,
+  baseStyles,
+  CommonStringsService,
+  CssHelpers,
+  ESC,
+  event,
+  EventEmitter,
+  property,
+  registerElementSafely,
+  UniqueId,
+} from '@clr/core/internal';
+import { html } from 'lit-element';
+import { CdsBaseFocusTrap } from '../internal/base/focus-trap.base';
+import { styles } from './modal.element.css';
+
+class ModalMixinClass extends CdsBaseFocusTrap {}
+
+applyMixins(ModalMixinClass, [UniqueId, CssHelpers]);
+
+/**
+ * Web component modal.
+ *
+ * ```typescript
+ * import '@clr/core/modal';
+ * ```
+ *
+ * ```html
+ * <cds-modal>
+ *   <cds-modal-header>My header</cds-modal-header>
+ *   <cds-modal-content>
+ *     <p>Lorem Ipsum</p>
+ *   </cds-modal-content>
+ *   <cds-modal-actions>
+ *       <cds-button>Ok</cds-button>
+ *   </cds-modal-actions>
+ * </cds-modal>
+ * ```
+ * @beta
+ * @element cds-modal
+ */
+export class CdsModal extends ModalMixinClass {
+  static get styles() {
+    return [baseStyles, styles];
+  }
+  @event() private closedChange: EventEmitter<boolean>;
+
+  /** If false, the modal will not render the close button.  */
+  @property({ type: Boolean })
+  closable = true;
+
+  /** Sets the overall height and width of the modal and icon based on value */
+  @property({ type: String })
+  size: 'default' | 'sm' | 'lg' | 'xl';
+
+  private ariaLabel = `aria-${this._idPrefix}${this._uniqueId}`;
+
+  render() {
+    return html`
+      <div class="private-host" cds-layout="p:sm p@md:lg">
+        <div class="modal-dialog" 
+              role="dialog" 
+              aria-modal="true" 
+              aria-labelledby="${this.ariaLabel}">
+          <div class="clr-sr-only">${CommonStringsService.keys.modalContentStart}</div>
+          <div class="modal-content" cds-layout="p:md">
+            <div cds-layout="horizontal gap:sm p-b:sm p-b@md:md">
+              <div cds-layout="align-stretch" id="${this.ariaLabel}"> 
+                  <span class="modal-title">
+                    <slot name="modal-header"></slot>
+                  </span>
+              </div>
+              <div cds-layout="align:right">
+                <slot name="modal-header-actions"></slot>   
+              </div>
+              ${
+                this.closable
+                  ? html`
+                <button
```

Comment:
We went back/forth on this a bit. Before the `cds-icon-button` was a separate component, I started out with `cds-button` but the default styling and functionality didn't quite fit well in modal and we discussed and changed to a plain button.  We are already addressing a11y concerns / features in this component. Do you see other benefits of using our button component here?

Reply:
At present, this button is being vertically centered based on line-height. There's also no real way of knowing how wide the button is, although I suspect it is just wide enough to enclose the icon.

There's CSS in `modal.element.scss` on lines 40, 41, and 52 that would not need to be there if we used an icon-button. Additionally, we could move `cds-icon-button.close` CSS into the icon button so it could be shared across components.